app/workers/stale_issue_marker.rb

```diff
@@ -23,49 +23,50 @@ class StaleIssueMarker
 Thank you for all your contributions!
   EOS
 
-  attr_reader :fq_repo_name
-
-  def perform(fq_repo_name)
-    @fq_repo_name = fq_repo_name
-    raise "The label #{STALE_LABEL_NAME} does not exist on #{fq_repo_name}" unless GithubService.valid_label?(fq_repo_name, STALE_LABEL_NAME)
-
-    GithubService.issues(fq_repo_name, :state => :open, :sort => :updated, :direction => :asc).each do |issue|
-      pinned = (issue.labels & PINNED_LABELS).present?
-
-      if issue.updated_at < stale_date && !pinned
-        if issue.pull_request?
-          closable_prs << issue
-        else
-          stale_issues << issue
-        end
+  def perform
+    GithubService.search_issues(*search_args).each do |issue|
+      if issue.pull_request?
+        close_pr(issue)
+      else
+        mark_as_stale(issue)
       end
     end
-
-    stale_issues.each do |issue|
-      next if issue.labels.include?(STALE_LABEL_NAME)
-      logger.info("[#{Time.now.utc}] - Marking issue #{fq_repo_name}##{issue.number} as stale")
-      issue.add_labels([STALE_LABEL_NAME])
-      issue.add_comment(STALE_ISSUE_MESSAGE)
-    end
-
-    closable_prs.each do |pr|
-      logger.info("[#{Time.now.utc}] - Closing stale PR #{fq_repo_name}##{pr.number}")
-      GithubService.close_pull_request(pr.fq_repo_name, pr.number)
-      pr.add_comment(CLOSABLE_PR_MESSAGE)
-    end
   end
 
   private
 
+  def search_args
+    query  = "is:open archived:false update:<#{stale_date.strftime('%Y-%m-%d')}"
+    query << " #{PINNED_LABELS.map { |label| %(-label:"#{label}") }.join(" ")}"
+    query << " #{enabled_repo_names.map { |repo| %(repo:"#{repo}") }.join(" ")}"
+
+    [query, {:sort => :updated, :direction => :asc}]
+  end
+
   def stale_date
     @stale_date ||= 6.months.ago
   end
 
-  def stale_issues
-    @stale_issues ||= []
+  def repo_has_label(repo)
+    unless GithubService.valid_label?(repo, STALE_LABEL_NAME)
+      raise "The label #{STALE_LABEL_NAME} does not exist on #{repo}"
+    end
+  end
+
+  def mark_as_stale(issue)
+    return if issue.labels.include?(STALE_LABEL_NAME)
+
+    repo_has_label(issue.fq_repo_name)
+
+    logger.info("[#{Time.now.utc}] - Marking issue #{issue.fq_repo_name}##{issue.number} as stale")
+    issue.add_labels([STALE_LABEL_NAME])
+    issue.add_comment(STALE_ISSUE_MESSAGE)
   end
 
-  def closable_prs
-    @closable_prs ||= []
+  def close_pr(pull_request)
+    repo_has_label(pull_request.fq_repo_name)
```

Comment:
The method name is weird to me...perhaps `validate_stale_label_presence`?

Reply:
Would `validate_repo_has_stale_label` work?  The `_presence` bit just feels weird to me.

#namingIsHard

Reply:
I went with my suggestion because I am self centered.  #dealWithIt

(in all seriousness, if you don't approve, I can change to your suggestion, just went with something to keep things moving)